app/controllers/test_reports_controller.rb

```diff
@@ -54,6 +59,28 @@ def update
 
   private
 
+  def create_job_tree(parent_jobs, children_jobs)
```

Comment:
(Same) It might be a good idea to move to job model as this line depends on job tables in in database.